File: drift-static-profiler/tests/fixtures/scala-play/app/services/OrdersService.scala
package services

import scala.concurrent.{ExecutionContext, Future}
import repos.{Order, OrdersRepository}

class OrdersService(repo: OrdersRepository)(implicit ec: ExecutionContext) {

  def createOrder(email: String, totalCents: Long, currency: Option[String]): Future[Long] = {
    val order = buildOrder(email, totalCents, currency)
    validate(order)
    repo.save(order)
  }

  def buildOrder(email: String, totalCents: Long, currency: Option[String]): Order = {
    Order(id = 0L, customerEmail = email, totalCents = totalCents, currency = currency.getOrElse("USD"))
  }

  def validate(order: Order): Unit = {
    if (order.totalCents <= 0L) {
      throw new IllegalArgumentException("totalCents must be positive")
    }
    if (!order.customerEmail.contains("@")) {
      throw new IllegalArgumentException("invalid customerEmail")
    }
  }
}

File: drift-static-profiler/tests/fixtures/kotlin-ktor/src/main/kotlin/com/example/services/OrdersService.kt
package com.example.services

import com.example.repos.Order
import com.example.repos.OrdersRepository

class OrdersService(private val repo: OrdersRepository) {

    fun createOrder(email: String, totalCents: Long, currency: String?): Long {
        val order = buildOrder(email, totalCents, currency)
        validate(order)
        return repo.save(order)
    }

    fun buildOrder(email: String, totalCents: Long, currency: String?): Order {
        return Order(
            id = 0L,
            customerEmail = email,
            totalCents = totalCents,
            currency = currency ?: "USD",
        )
    }

    private fun validate(order: Order) {
        if (order.totalCents <= 0L) {
            throw IllegalArgumentException("totalCents must be positive")
        }
        if (!order.customerEmail.contains("@")) {
            throw IllegalArgumentException("invalid customerEmail")
        }
    }
}
